# Joint state controller
joint_state_controller:
  publish_rate: 125
  type: joint_state_controller/JointStateController

# Gripper controller
gripper_controller:
  type: position_controllers/GripperActionController
  joint: robotiq_85_left_knuckle_joint
  action_monitor_rate: 20
  goal_tolerance: 0.002
  max_effort: 100
  stall_velocity_threshold: 0.001
  stall_timeout: 1.0

# Trajectory controller
trajectory_controller:
  type: "position_controllers/JointTrajectoryController"
  joints:
    - j1
    - j2
    - j3
    - j4
    - j5
    - j6
  constraints:
    # Default is 0
    goal_time: 0.1
    # Default is 0.01
    stopped_velocity_tolerance: 0.01
    j1:
      trajectory: 0
      goal: 0.01
    j2:
      trajectory: 0
      goal: 0.01
    j3:
      trajectory: 0
      goal: 0.01
    j4:
      trajectory: 0
      goal: 0.01
    j5:
      trajectory: 0
      goal: 0.01
    j6:
      trajectory: 0
      goal: 0.01
#Position controller
j1:
  type: position_controllers/JointPositionController
  joint: j1
j2:
  type: position_controllers/JointPositionController
  joint: j2
j3:
  type: position_controllers/JointPositionController
  joint: j3
j4:
  type: position_controllers/JointPositionController
  joint: j4
j5:
  type: position_controllers/JointPositionController
  joint: j5
j6:
  type: position_controllers/JointPositionController
  joint: j6
